File: Client_Root/Client/Assets/Scripts/Room/Characters/MisterBae.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class MisterBae : ICharacter
{
#region ICharacter
    protected override void CreateUI()
    {
        m_CharacterUI = (Instantiate(Resources.Load("CharacterUI/MisterBaeUI")) as GameObject).GetComponent<MisterBaeUI>();
    }

    public override void Initialize(params object[] arrParam)
    {
        m_DefaultStat = m_CurrentStat = (Stat)arrParam[0];
    }

    public override void Idle()
    {
        StopAllBehaviors();

        StartBehavior(new IdleBehavior(this, "WAIT01", m_LastBehavior is IdleBehavior));
    }

    public override void Stop()
    {
        StopAllBehaviors();

        StartBehavior(new StopBehavior(this));
    }

    public override void Move(LinkedList<Node> listPath, float fEventTime)
    {
        StopAllBehaviors();

        StartBehavior(new MoveBehavior(this, listPath, "RUN00_F", m_LastBehavior is MoveBehavior, fEventTime));
    }

    public override void Skiil(object data)
    {
        
    }

    public override void Emotion()
    {
        
    }
#endregion
}

File: Server_Root/Server/RoomServer/Entity/Entities/Character/Characters/MisterBae.cpp
#include "stdafx.h"
#include "MisterBae.h"
#include "../../../../Behavior/Behaviors/Idle.h"
#include "../../../../Behavior/Behaviors/Move.h"

MisterBae::MisterBae(int nID) : ICharacter(nID)
{
	InitializeBehavior();
}

MisterBae::~MisterBae()
{
}

void MisterBae::InitializeBehavior()
{
	m_listBehavior.push_back(new Idle(this));
	m_listBehavior.push_back(new Move(this));
}

void MisterBae::Initialize()
{
}
